### PR TITLE
FIX-#1869: index sort for count(level=...)

### LINKED ISSUE #1869: Pandas count(level=...) sorts index while Modin doesn't do this
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **Modin version** (`modin.__version__`): 0.7.3+281.g4d80bd8
- **Python version**: 3.8.3
- **Code we can use to reproduce**:
```
import pandas
import modin.pandas as pd
import numpy as np

from modin.pandas.test.utils import test_data

df_modin, df_pandas = pd.DataFrame(test_data["dense_nan_data"]), pandas.DataFrame(test_data["dense_nan_data"])
rows_number = len(df_modin.index)
level_0 = np.random.choice([x for x in range(10)], rows_number)
level_1 = np.random.choice([x for x in range(10)], rows_number)
index = pandas.MultiIndex.from_arrays([level_0, level_1])

df_modin.index = index
df_pandas.index = index

print("df_modin.count(level=0).index", df_modin.count(level=0).index) # Int64Index([8, 2, 4, 0, 5, 6, 3, 7, 9, 1], dtype='int64')
print("df_pandas.count(level=0).index", df_pandas.count(level=0).index) # Int64Index([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], dtype='int64')
```

### PR DESCRIPTION
**Pull request by amyskov**
Signed-off-by: Alexander Myskov <alexander.myskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1869  
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -751,7 +751,9 @@ def count(self, axis=0, level=None, numeric_only=False):
                 # error thrown by pandas
                 raise TypeError("Can only count levels on hierarchical columns.")
 
-            return self._handle_level_agg(axis, level, "count")
+            result = self._handle_level_agg(axis, level, "count")
+            result.sort_index(inplace=True)
+            return result
```
Comment (dchigarev) [inline modin/pandas/base.py:756]: `_handle_level_agg` has an internal `sort` parameter which it passes to groupby, maybe we should make that parameter external for `_handle_level_agg` and will pass `sort=True` for `count` besides sorting the result?

Comment (amyskov) [reply modin/pandas/base.py]: Sounds reasonable, implemented in the new commit.

**Comment by dchigarev** [comment]
is there are any other operations that uses `_handle_agg_level` and their results have sorted indices at pandas, but in modin it does not?

**Comment by amyskov** [comment]
> is there are any other operations that uses `_handle_agg_level` and their results have sorted indices at pandas, but in modin it does not?

For now `test_index_order` tests cover all cases of using of `_handle_level_agg` function, so i think there are no such functions.

**Comment by amyskov** [comment]
@dchigarev your approve was dismissed during rebase, can you make approve once more time?

**Pull request merged**